miniapp/src/main/java/com/rakuten/tech/mobile/miniapp/storage/MiniAppSecureStorage.kt

```diff
@@ -0,0 +1,249 @@
+            close()
+        }
+        onSuccess(deSerializeItems(content))
+    } catch (e: Exception) {
+        onFailed(MiniAppStorageError.ioError)
+    }
+
+    @VisibleForTesting
+    @Suppress("SwallowedException", "TooGenericExceptionCaught", "ReturnCount")
+    private fun readFromEncryptedFile(miniAppId: String): Map<String, String>? {
+        try {
+            if (isStorageAvailable(miniAppId)) {
+                val encryptedFile = EncryptedFile.Builder(
+                    activity,
+                    File(secureStorageBasePath, "$miniAppId.txt"),
+                    MasterKey.Builder(activity).setKeyScheme(MasterKey.KeyScheme.AES256_GCM)
+                        .build(),
+                    EncryptedFile.FileEncryptionScheme.AES256_GCM_HKDF_4KB
+                ).build()
+
+                val inputStream = encryptedFile.openFileInput()
+                val byteArrayOutputStream = ByteArrayOutputStream()
+                var nextByte: Int = inputStream.read()
+                while (nextByte != -1) {
+                    byteArrayOutputStream.write(nextByte)
+                    nextByte = inputStream.read()
+                }
+
+                val plaintext: ByteArray = byteArrayOutputStream.toByteArray()
+                val jsonToRead = plaintext.toString(Charsets.UTF_8)
+                return deSerializeItems(jsonToRead)
+            } else {
+                return null
+            }
+        } catch (e: Exception) {
+            return null
+        }
+    }
+
+    fun deleteSecureStorage(
+        miniAppId: String,
+        onSuccess: () -> Unit,
+        onFailed: (MiniAppStorageError) -> Unit
+    ) {
+        if (isStorageAvailable(miniAppId)) {
+            scope.launch {
+                val file = File(secureStorageBasePath, "$miniAppId.txt")
+                file.delete()
+                if (!file.exists()) {
+                    onSuccess()
+                } else {
+                    onFailed(MiniAppStorageError.failedDeleteError)
+                }
+            }
+        } else {
+            onFailed(MiniAppStorageError.unavailableStorage)
+        }
+    }
+
+    fun deleteSecureStorageItems(
+        miniAppId: String,
+        keySet: Set<String>,
+        onSuccess: (Map<String, String>) -> Unit,
+        onFailed: (MiniAppStorageError) -> Unit
+    ) {
+        scope.launch {
+            storageState.postValue(StorageState.LOCK)
+            if (isStorageAvailable(miniAppId)) {
+                val storedItems = readFromEncryptedFile(miniAppId)
+                storedItems?.let { items ->
+                    val filterItems = items.filter { !keySet.contains(it.key) }
+                    writeToEncryptedFile(
+                        miniAppId,
+                        serializedItems(filterItems),
+                        onSuccess,
+                        onFailed
+                    )
+                } ?: kotlin.run {
+                    onFailed(MiniAppStorageError.emptyStorage)
```

Comment:
I think we shouldn't return an error in this case. We can just return a normal success because the storage is already cleared.